### PR TITLE
Do not add schema / class name mapping where custom mapping exists

### LINKED ISSUE #13150: Spring generates duplicated JsonSubTypes
Hello everyone,

When I use the openapi-generator with a discriminator and mapping, it generates two @JsonSubTypes

```
discriminator:
    propertyName: type
    mapping:
         foo: "#components/schemas/Foo"
```

Generates something like this

```
@JsonSubTypes.Type(value = Foo.class, name = "Foo")
@JsonSubTypes.Type(value = Foo.class, name = "foo")
```

Notice the duplication. The Foo is the class name of the class, and the lowercase foo is my mapping. Wtf is going on lol. This doesn't seem right.

The `Foo` class is defined something like

```
Foo:
    allOf:
         - $ref: "#components/schemas/someParent"
         - type: object
            properties:
                type:
                    type: string
```

I am using version 6.0.1 and the schema version is `openapi: '3.0.0'`

Here's the full spec to generate the incorrect thing:
```
openapi: '3.0.0'
info:
  version: '1.0.0'
  title: 'FooService'
paths:
  /parent:
    put:
      summary: put parent
      operationId: putParent
      parameters:
        - name: name
          in: path
          required: true
          description: Name of the account being updated.
          schema:
            type: string
      requestBody:
        description: The updated account definition to save.
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Parent'
      responses:
        '200':
          $ref: '#/components/responses/Parent'
components:
  schemas:
    Parent:
      type: object
      description: Defines an account by name.
      properties:
        name:
          type: string
          description: The account name.
        type:
          type: string
          description: The account type discriminator.
      required:
        - name
        - type
      discriminator:
        propertyName: type
        mapping:
          foo: '#/components/schemas/Foo'

    Foo:
      allOf:
        - $ref: "#/components/schemas/Parent"
        - type: object
          properties:
            fooType:
              type: string 
  responses:
    Parent:
      description: The saved account definition.
      content:
        application/json:
          schema:
            $ref: '#/components/schemas/Parent'
```

I generated this by running
`java -jar openapi-generator-cli.jar generate -i schema.yaml -o foo -g spring`

### PR DESCRIPTION
**Pull request by wing328**
- fix #13150  by cherry picking #13815
- to close https://github.com/OpenAPITools/openapi-generator/issues/14917
- avoid unnecessary import

credits: @bernie-schelberg-mywave

cc @OpenAPITools/generator-core-team 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by WannabeSoftwareEngineer** [comment]
Hey @wing328,
thank you for all your great work. Is this pr expected to make to 6.5.0?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java [hunk 1]
```diff
@@ -734,7 +734,9 @@ private Set<String> rewriteImports(Set<String> originalImports, boolean isModel)
                 resultImports.add(i);
             } else if (importMapping().containsKey(modelImport)) {
                 resultImports.add(importMapping().get(modelImport));
-            } else if (modelImport.startsWith("package:")) {
+            } else if ("dart:core".equals(modelImport)) { // import dart:core directly
+                resultImports.add(modelImport);
```
Comment (ahmednfwela) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:738]: `dart:core` isn't the only one, you can also get dart:convert, dart:typed_data, dart:async, etc...

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:738]: Is there a complete list that I can include?

Comment (ahmednfwela) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:738]: well in dart you only get 3 types of imports:

1. `package:`
2. `dart:`
3. relative

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:738]: ok, i'll do a check to see if `modelImport` starts with `dart:`

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java]: done viia https://github.com/OpenAPITools/openapi-generator/pull/14984/commits/7027fa9f39b1de90406ff3bd103fce121c73efcd

**Pull request merged**